src/ctap/customization.rs

```diff
@@ -19,6 +19,15 @@
 
 use crate::ctap::data_formats::{CredentialProtectionPolicy, EnterpriseAttestationMode};
 
+#[derive(Clone)]
+pub struct Customization {
+    pub max_msg_size: usize,
+}
+
+pub const DEFAULT_CUSTOMIZATION: Customization = Customization {
+    max_msg_size: MAX_MSG_SIZE,
```

Comment:
@ia0 WDYT about the structure in this file? Should we keep the constants and comments in original place and use this style, or do you think we should put all comments and values directly inside this struct and delete the original consts?

Reply:
Good question. The way I see it:

```rust
// src/api/customization.rs
pub trait Customization {
    /// "Documentation of the constant"
    fn the_constant(&self) -> TheType;  // IMPORTANT: This is &self
    // ... and so on ...
}

pub struct Impl {
    pub the_constant: TheType,
    // ...
}

impl Customization for Impl {
    fn the_constant(&self) -> TheType { self.the_constant }
    // ...
}

pub const DEFAULT: Impl = Impl {
    the_constant: THE_DEFAULT,
    // ...
}

// src/env/mod.rs
pub trait Env {
    // ...
    type Customization: Customization;

    // ...
    fn customization(&mut self) -> &mut Customization;  // IMPORTANT: This is &mut self
    // I think we should always use `&mut env` and thus it's not a huge restriction to have &mut self here.
    // Also it's needed to modify the customization in tests. Because I think it's more convenient to just create
    // the Test Env without arguments, then modify it, and then use it. Rather than pass a customization at
    // creation time.
}
```

So `src/ctap/customization.rs` would disappear and be moved to `src/api`. Since you're doing this incrementally (constant by constant) which is good, this migration will occur incrementally.

Reply:
Looks good, except `fn customization(&mut self) -> &mut Customization;` won't work because we are returning a ref to the constant DEFAULT in `TockEnv`. Specifically customization can't be changed in prod.
So the trait method should be kept as `fn customization(&self) -> &Customization;`, but expose another method `customization_mut` in `TestEnv`. 

I didn't use trait at first because in my imagination that impls will always be just containing exact identical fields as the constants we want to access. But using trait makes sense too since some constants are dependent on each other and we can have less fields (than num of constants) in TestCustomization. This way we can enforce valid customization settings when fuzzing (if we want).

Reply:
We will have to enforce valid settings there. See the unit test in the original `customization.rs`, this has to be checked.

Reply:
Right, I did not think of the problem returning a constant in `TockEnv`. The `customization_mut` sounds good.